### PR TITLE
[travis] Revert tests cancellations and don't fail if memcached is not avail

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

I don't understand how travis gets its $TRAVIS_COMMIT, but this should make things stable again.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
So, I reverted #14047 and added a check to not fail if memcached can't be installed.
If someone has a better idea, I'd close this PR happily.
For now, I'm stuck with this.

File: .travis.yml [hunk 1]
```diff
@@ -26,15 +26,12 @@ env:
         - SYMFONY_DEPRECATIONS_HELPER=weak
 
 before_install:
-    - if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then git fetch origin "refs/pull/$TRAVIS_PULL_REQUEST/merge"; else git fetch origin "$TRAVIS_BRANCH"; fi;
-    - if [[ "$TRAVIS_COMMIT" != `git rev-parse FETCH_HEAD` ]]; then echo "Pull request or branch commit hash has changed, aborting!"; exit 1; fi;
```
Comment (xabbuh) [inline .travis.yml]: Did you figure out what actually happened on Travis to make this fail?

Comment (nicolas-grekas) [inline .travis.yml]: Somehow: they have $TRAVIS_COMMIT hash that do not exist anymore in the main repos. I suspect this is related to github doing merges all the time in `refs/pull/*/merge`, eveytime we merge something in the target branch.

**Comment by nicolas-grekas** [comment]
Travis should be back to green now that this is merged, please rebase your PRs pull-requesters

**Pull request merged**